libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt

```diff
@@ -49,6 +56,10 @@ class RustSyncService(
         roomListService
             .stateFlow()
             .map(RoomListServiceState::toSyncState)
+            .onEach { state ->
+                Timber.v("Sync state=$state")
+                isSyncing.set(state == SyncState.Syncing)
+            }
```

Comment:
Is this properly mapping `RoomListServiceState::{Init, SettingUp}` to `Syncing`? If not, this is wrong, and may cause a `true -> false` transition (if the room list service was explicitly started, and then it's moving on to `SettingUp`)

Reply:
```kotlin
internal fun RoomListServiceState.toSyncState(): SyncState {
    return when (this) {
        RoomListServiceState.INIT,
        RoomListServiceState.SETTING_UP -> SyncState.Idle
        RoomListServiceState.RUNNING -> SyncState.Syncing
        RoomListServiceState.ERROR -> SyncState.InError
        RoomListServiceState.TERMINATED -> SyncState.Terminated
    }
}
```

Only `Running` is mapped to `Syncing`, so I *think* it should be fine?

Reply:
> you start the sliding sync, so the atomic gets set to true

My bad, I didn't realise we were still using `compareAndSet` there. @ganfra maybe we should re-think this?